src/box/lua/console.lua

```diff
@@ -985,6 +986,9 @@ local function client_handler(client, _peer)
     state:print(string.format("%-63s\n%-63s\n",
         "Tarantool ".. version.." (Lua console)",
         "type 'help' for interactive help"))
+    box.ctl.on_shutdown(function()
+        state.running = false
```

Comment:
Don't you need to wake up the console fiber here?

Reply:
Change this hunk to:
```diff
@@ -985,7 +986,14 @@ local function client_handler(client, _peer)
     state:print(string.format("%-63s\n%-63s\n",
         "Tarantool ".. version.." (Lua console)",
         "type 'help' for interactive help"))
+    local on_shutdown = function()
+        state.running = false
+        fiber.cancel(state.fiber)
+    end
+    state.fiber = fiber.self()
+    box.ctl.on_shutdown(on_shutdown)
     repl(state)
+    box.ctl.on_shutdown(nil, on_shutdown)
     session_internal.run_on_disconnect()
 end
```